### ISSUE TITLE
Flaky test: Windows test_configuration_loader.deleting_parent_of_missing_file_is_not_detected_as_a_change

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#479]
Windows MSVC Release
https://github.com/quick-lint/quick-lint-js/runs/4041577979?check_suite_focus=true

```
1: [ RUN      ] test_configuration_loader.deleting_parent_of_missing_file_is_not_detected_as_a_change
1: D:\a\quick-lint-js\quick-lint-js\src\change-detecting-filesystem-win32.cpp:142: internal check failed in handle_event: false
1: quick-lint-js crashed. Please report this bug here:
1: https://quick-lint-js.com/crash-report/
1/2 Test #1: quick-lint-js-test ...............***Failed    0.37 sec
```

**Comment by strager** [comment]
I couldn't reproduce this on my machine after 100k repetitions of the test alone and 1k repetitions of all test_configuration_loader.* tests.

**Comment by strager** [comment]
Got the failure again, this time with a bit more info: https://github.com/quick-lint/quick-lint-js/runs/4101594854?check_suite_focus=true
```
1: [ RUN      ] test_configuration_loader.deleting_parent_of_missing_file_is_not_detected_as_a_change
1: error: change_detecting_filesystem_win32 received unexpected error: 3 (number_of_bytes_transferred=24)
1: D:\a\quick-lint-js\quick-lint-js\src\change-detecting-filesystem-win32.cpp:146: internal check failed in handle_event: false
1: quick-lint-js crashed. Please report this bug here:
1: https://quick-lint-js.com/crash-report/
1/2 Test #1: quick-lint-js-test ...............***Failed    0.39 sec
```

**Comment by strager** [comment]
Fixed in cb0f0a85c14249d09f8629687a9c2c09ba2e996a.
